File: Chapter 04/C#/Section_4_2_Downward_Dependency/After/OrderService.cs
namespace Chapter04.Section_4_2.After
{
	public class OrderService
	{
	    private readonly IOrderRepository _repo; // Depends on abstraction 
	    public OrderService(IOrderRepository repo) 
        { 
            _repo = repo; 
        }
	    public void SaveOrder(Order order)
	    {
	        // Calls DOWNWARDS via interface 
	        _repo.Save(order);
	    }
	}
}

File: Chapter 04/JS/section_4_2_downward_dependency/after/orderService.js
/**
 * BUSINESS LOGIC LAYER.
 * ARCHITECTURE NOTE: No 'IOrderRepository' file is needed. 
 * JS uses Duck Typing. As long as the 'repo' object has a 
 * .save() method, the code will execute successfully.
 */
class OrderService {
    constructor(repo) { 
        this.repo = repo; 
    }

    saveOrder(order) {
        // Calls DOWNWARDS. The service doesn't care 
        // if this is SQL, NoSQL, or a Mock.
        this.repo.save(order);
    }
}

module.exports = OrderService;